### PR TITLE
(GB)(E2E) E2E tests fixes for releasing GB 16.7.x to WPCOM 

### PR DESCRIPTION
**Pull request by fullofcaffeine**
## Testing Instructions

* Checks should pass
* The GB edge E2Es should pass from this branch (AT and simple)

Tracking issue: https://github.com/Automattic/wp-calypso/issues/81961

### CODE REVIEW

File: packages/calypso-e2e/src/lib/pages/editor-page.ts [hunk 1]
```diff
@@ -280,8 +280,9 @@ export class EditorPage {
 			noSearch: noSearch,
 		} );
 
-		const blockHandle =
-			await this.editorGutenbergComponent.getSelectedBlockElementHandle( blockEditorSelector );
+		const blockHandle = await this.editorGutenbergComponent.getSelectedBlockElementHandle(
```
Comment (fullofcaffeine) [inline packages/calypso-e2e/src/lib/pages/editor-page.ts]: Hmm, linting/formatting changes only.

File: packages/calypso-e2e/src/lib/pages/editor-page.ts [hunk 2]
```diff
@@ -330,8 +331,9 @@ export class EditorPage {
 		await openInlineInserter( await this.editor.canvas() );
 		await this.addBlockFromInserter( blockName, this.editorInlineBlockInserterComponent );
 
-		const blockHandle =
-			await this.editorGutenbergComponent.getSelectedBlockElementHandle( blockEditorSelector );
+		const blockHandle = await this.editorGutenbergComponent.getSelectedBlockElementHandle(
```
Comment (fullofcaffeine) [inline packages/calypso-e2e/src/lib/pages/editor-page.ts]: ditto above

File: test/e2e/specs/editor/editor__post-basic-flow.ts [hunk 3]
```diff
@@ -86,7 +86,11 @@ describe( DataHelper.createSuiteTitle( 'Editor: Basic Post Flow' ), function ()
 
 	describe( 'Jetpack features', function () {
 		it( 'Open Jetpack settings', async function () {
-			await editorPage.openSettings( 'Jetpack' );
+			await editorPage.openEditorOptionsMenu();
```
Comment (fullofcaffeine) [inline test/e2e/specs/editor/editor__post-basic-flow.ts:89]: This is a workaround to click the "Jetpack" item in the dropdown instead of the pinner Jetpack item button in the toolbar. The Jetpack pinned button is not present in most mobile sizes in Gutenberg 16.7, see: p9Jlb4-93O-p2.

An idea might be to refactor the "openSettings" method just always to use the "options" dropdown to avoid the whole problem with pinned buttons. The plugin buttons are always rendered in the options dropdown, regardless of the viewport size (see below). 

The TL;DR is: this logic works, but we might need to think more about abstracting this better. It's also important to note that with this change, the test opens the Jetpack settings by first going to the "Options" menu in the _Desktop_ viewport too. I left it as is because, for all intents and purposes, it works, but let me know if you want to keep clicking the jetpack pinned button instead.

File: packages/calypso-e2e/src/lib/components/editor-toolbar-component.ts [hunk 4]
```diff
@@ -446,7 +446,10 @@ export class EditorToolbarComponent {
 
 		// To support i18n tests.
 		const translatedTargetName = await this.translateFromPage( 'Options' );
-		const button = editorParent.getByRole( 'button', {
+		// Narrowing down to the "Editor top bar" is needed because it might conflict with
+		// the options button for the block toolbar, causing a strict-mode violation error
+		// due to duplicate elements on the page.
+		const button = editorParent.getByLabel( 'Editor top bar' ).getByRole( 'button', {
```
Comment (fullofcaffeine) [inline packages/calypso-e2e/src/lib/components/editor-toolbar-component.ts:452]: I had to add the `getByLabel` here because of the following error:

```
    locator.getAttribute: Error: strict mode violation: locator('body.block-editor-page').getByRole('button', { name: 'Options', exact: true }) resolved to 2 elements:                                             
        1) <button type="button" aria-haspopup="true" aria-label="O…>…</button> aka getByLabel('Editor top bar').getByLabel('Options')                                                                              
        2) <button id=":r28:" type="button" tabindex="-1" data-comm…>…</button> aka getByLabel('Block tools').getByLabel('Options')        
```

File: test/e2e/specs/editor/editor__post-basic-flow.ts [hunk 5]
```diff
@@ -86,7 +86,11 @@ describe( DataHelper.createSuiteTitle( 'Editor: Basic Post Flow' ), function ()
...
+			const page = await editorPage.getEditorParent();
+
+			const button = await page.locator( 'button:has(span:has-text("Jetpack"))' );
```
Comment (dpasque) [inline test/e2e/specs/editor/editor__post-basic-flow.ts]: **suggestion (non-blocking)**: If you're open to it, I would recommend the React Testing Library style syntax here, like `getByRole( 'menuitemcheckbox', { name: "Jetpack" } );` We try to use that wherever possible! 😄

**Review by dpasque — APPROVED** [review]
One small suggestion! Thank you for making the fix/updates 😄

File: test/e2e/specs/editor/editor__post-basic-flow.ts (hunk 5 above)
Comment (fullofcaffeine) [reply test/e2e/specs/editor/editor__post-basic-flow.ts]: Ahh, sure! I'll try it out :)
